onnxruntime/core/providers/tensorrt/tensorrt_execution_provider.cc

```diff
@@ -555,6 +555,46 @@ Status TensorrtExecutionProvider::SetComputeStream(void* stream) {
   return Status::OK();
 }
 
+// Check the graph is the subgraph of control flow op
+bool TensorrtExecutionProvider::IsSubGraphOfControlFlowOp(const GraphViewer& graph) const {
+  if (graph.IsSubgraph()) {
+    const auto& node = graph.ParentNode();
+    if (control_flow_op_set_.find(node->OpType()) != control_flow_op_set_.end()) {
+      return true;
+    }
+  }
+  return false;
+}
+
+
+// Check whether all the nodes of the graph are assigned to specific ep 
+bool TensorrtExecutionProvider::AllNodesAssignedToSpecificEP(const GraphViewer& graph, const std::string& provider_type) const {
```

Comment:
this method doesn't seem TRT EP specific.  consider moving out to GraphUtils in a future refactor?

Reply:
yes, make sense to move this function to GraphUtils. Will do this in the future refactor.